src/NUnitEngine/nunit.engine/ITestAgency.cs

```diff
@@ -34,5 +34,11 @@ public interface ITestAgency
         /// </summary>
         /// <param name="agent"></param>
         void Register(ITestAgent agent);
+
+        /// <summary>
+        /// Unregisters an agent with an agency
+        /// </summary>
+        /// <param name="agent"></param>
+        void Unregister(ITestAgent agent);
```

Comment:
The guid seems best to me. No reason to pass more info to the database than it really needs.

That said, let's first make sure we are doing enough with this. Have you checked what happens to a process runner that's holding a test agent and runner when this de-registration happens?